framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2061,6 +2069,75 @@ protected void adaptGetClassReturnTypeToReceiver(
         classWildcardArg.setExtendsBound(newBound);
     }
 
+    /**
+     * Return the element type of {@code expression}. This is usually the type of {@code
+     * expression.itertor().next()}. If {@code expression} is an array, it is the component type of
+     * the array.
+     *
+     * @param expression an expression whose type is an array or implements {@link Iterable}
+     * @return the type of {@code expression.itertor().next()} or if {@code expression} is an array,
+     *     the component type of the array.
+     */
+    public AnnotatedTypeMirror getIteratorElementType(ExpressionTree expression) {
```

Comment:
Should this be `getIterableElementType`?  That indicates the type of the argument more precisely.  I'm OK with the current name, too.

Reply:
Yes, I made that change.